src/bin/utils/processors.ts

```diff
@@ -219,7 +220,7 @@ async function processClientOptions(
     if (clientPort == null) clientPort = statusInfo.data.clientPort;
   }
   return {
-    nodeId,
+    nodeId: nodesUtils.decodeNodeId(nodeId)!,
```

Comment:
This makes the assumption that the node ID from `Status` will be valid. Can we ever have a `Status` file that contains invalid data?

Reply:
It's possible. but not under normal operation.

Reply:
We would like to have `StatusInfo` contain a legitimate `NodeId`. To do this, we would need to ensure that during the `readStatus` call, it should returning a proper `NodeId` within `StatusInfo`.

So `StatusLive` should be using `NodeId` and `NodeIdEncoded`, and the `readStatus` and `writeStatus` should be decoding and encoding the node Id respectively.

To ensure that we have a proper `NodeId`, you want to have the `nodesUtils.decodeNodeId` function check if the resulting length is 32 bytes. If that's incorrect, then it should return `undefined`. This gives us a light weight validation upon decoding the node id.

In terms of validation, it should always occur at the boundaries of the app. In this line inside `processClientOptions`, the correct boundary is actually the `readStatus` method of `Status`, not this particular function. So if the node id is invalid, it will throw `ErrorStatusParse`.

Now there's a separate problem to deal with which the tests have dummy node ids that don't fit the spec for what a node id is. Right now we have to keep with a set size for node ids, which is 32 bytes. This is because the node graph cannot work with arbitrary sized node ids. These tests should be updated to use a dummy node id generator which exist in `tests/utils.ts`.

```ts
const nodeId: NodeId = IdInternal.create(await keysUtils.getRandomBytes(32));
```

So tests should be updated to use legitimate node ids like above, and we can see that `StatusInfo` is a place where `NodeId` has to be used instead of `NodeIdEncoded`.

Reply:
All relevant tests have been updated to use `generateRandomNodeId()`. I haven't updated  Status to do the verification directly yet.

Reply:
Fixed Status to use `NodeId`. had to add it's own reviver and replacer to make it work.